### DISCUSSION TITLE
a few verions of generation

### DISCUSSION THREAD

**Discussion by wiiiktor** [guidance-ai/guidance#523] [Ideas]
Let's assume LLM generated an idea "Let's organize a costumes party!" and now I want it to follow along, but I only want to accept ideas for party in Wild West, Jungle, Ancient Egypt and Mars (I know, it can be enforced with select(), but this is just an example). So, I instruct the model"Give me plan for such a party event, to organize everything smoothly", but once I check it is not one of the four options (Wilde West or the like), I want to reset the model and start again from the "Let's organize a costumes party!" moment. Is this possible? Can I "rewind the tape"?

**Comment by fullstackwebdev** [comment]
just store a copy of lm variable as lm_before, and use it when you want to go back.
